tests/unit/test_request.py

```diff
@@ -0,0 +1,21 @@
+import unittest
+
+from jina.clients.python.request import _generate
+from tests import JinaTestCase
+
+
+class MyTestCase(JinaTestCase):
+
+    def test_request_generate(self):
+
+        def random_lines(num_lines):
+            for j in range(num_lines):
+                yield "https://github.com 'i\'m dummy doc %d'" % j
+
+        req = _generate(random_lines(100))
+
+        self.assertEqual(len(list(req)), 1)
```

Comment:
why does it expect it to just return 1 element?

Reply:
Is it because of the `batch_size`? shouldn't it return 100 documents?

Reply:
yes if we pass `batch_size=0` than it's return 1 req with all docs and if we pass `batch_size=10` than it's return 10 req with 10 docs each

Reply:
so can you pass `batch_size=100`? I think this way seems a more explicit test

Reply:
yep

Reply:
done

Reply:
but the assertion should not return 100 elements?

Reply:
no no 
like if `batch_size=100` than it's return 1 req with 100 docs
if we pass `batch_size=100` than it's return 10 req with 10 docs each

Reply:
ah, then can we `assert` that the request contains 100 documents?

Reply:
yes, like that
`req = _generate(data=random_lines(100), batch_size=100)`
len of req list is 1 but on that req we have 100 docs. so, i'm checking with len of req's list